tsl/test/shared/expected/gapfill.out

```diff
@@ -3292,11 +3294,11 @@ SELECT time_bucket_gapfill('2 month'::interval, ts, 'Europe/Berlin', '2000-01-01
      time_bucket_gapfill      
  Fri Dec 31 15:00:00 1999 PST
  Tue Feb 29 15:00:00 2000 PST
- Sat Apr 29 15:00:00 2000 PDT
- Thu Jun 29 15:00:00 2000 PDT
- Tue Aug 29 15:00:00 2000 PDT
- Sun Oct 29 15:00:00 2000 PST
- Fri Dec 29 15:00:00 2000 PST
+ Sun Apr 30 15:00:00 2000 PDT
+ Fri Jun 30 15:00:00 2000 PDT
+ Thu Aug 31 15:00:00 2000 PDT
+ Tue Oct 31 15:00:00 2000 PST
+ Sun Dec 31 15:00:00 2000 PST
```

Comment:
I see that this is the last day of the month rather than the 29:th, but is correct? When I do a test I get the following result:

```sql
mats=# select day::date, day::date + '1 month'::interval, day::date + '2 month'::interval from (values ('2023-02-28'), ('2023-08-31'), ('2023-01-29')) as d(day);
    day     |      ?column?       |      ?column?       
------------+---------------------+---------------------
 2023-02-28 | 2023-03-28 00:00:00 | 2023-04-28 00:00:00
 2023-08-31 | 2023-09-30 00:00:00 | 2023-10-31 00:00:00
 2023-01-29 | 2023-02-28 00:00:00 | 2023-03-29 00:00:00
(3 rows)
```

Reply:
I don't understand why is it not correct? If these are montly buckets, should they not start/end on the last day of the month for every bucket?

Even your example shows that here:
`2023-08-31 | 2023-09-30 00:00:00 | 2023-10-31 00:00:00`

Reply:
Sorry, the case above is actually correct so it is not a good illustration of the problem I was wondering about.

I implemented this in PL/pgSQL to convince myself this works the way it should and it seems like that.

```sql
CREATE OR REPLACE FUNCTION advance_offset(start date, steps int, ival interval)
RETURNS date
LANGUAGE plpgsql AS $$
DECLARE
    off interval;
    count int := steps;
BEGIN
    off := ival;
    LOOP
        off := off + ival;
	count := count - 1;
	IF count = 0 THEN
	    EXIT;
	END IF;
    END LOOP;
    RETURN start + off;
END
$$;

CREATE OR REPLACE FUNCTION advance_date(start date, steps int, ival interval)
RETURNS date
LANGUAGE plpgsql AS $$
DECLARE
    ts date;
    count int := steps;
BEGIN
    ts := start;
    LOOP
        ts := ts + ival;
	count := count - 1;
	IF count = 0 THEN
	    EXIT;
	END IF;
    END LOOP;
    RETURN ts;
END
$$;
```

Which then gives the following result:

```sql
mats=# SELECT start,
       advance_date(start::date, 10, '1 month'::interval),
       advance_offset(start::date, 10, '1 month'::interval),
       CAST(start::date + 10 * '1 month'::interval AS date)
  FROM (VALUES ('2023-01-28'), ('2023-01-29'), ('2023-01-30')) AS tbl(start);
    start    | advance_date | advance_offset |    date    
------------+--------------+----------------+------------
 2023-01-28 | 2023-11-28   | 2023-12-28     | 2023-11-28
 2023-01-29 | 2023-11-28   | 2023-12-29     | 2023-11-29
 2023-01-30 | 2023-11-28   | 2023-12-30     | 2023-11-30
(3 rows)
```

The middle one is the original computation, which illustrates the problem, and the last one is the new implementation. I will approve the patch, but it would be good if you could use some extra dates to the test similar to the ones I have here so that we make sure that it works both when the date lands on the "last day of the month" as part of the computation and when the starting date is not the last day of the month (and the end result should not end on the last day of the month).